### ISSUE TITLE
Will fit_generator circle input file?

### ISSUE THREAD

**Issue by MaratZakirov** [fchollet/keras#5233]
I have very huge file of words which I want to convert to vectors and put them to learn online. As I read other issues the most "correct" way to do so is to use model.fit_generator. But I am in doubt will fit_generator circle my huge file by itself? And second question what to put  to samples_per_epoch to make fit_generator use all data from file?

**Comment by MaratZakirov** [comment]
Ok I should use while(1) in generator and use generator for validation set also. Thank you everyone for help.
